InternalTestApp/PrebidMobileDemoRendering/ViewControllers/Adapters/Prebid/CustomRenderer/CustomRendererBannerController.swift

```diff
@@ -0,0 +1,249 @@
+/*   Copyright 2018-2021 Prebid.org, Inc.
+
+ Licensed under the Apache License, Version 2.0 (the "License");
+ you may not use this file except in compliance with the License.
+ You may obtain a copy of the License at
+
+ http://www.apache.org/licenses/LICENSE-2.0
+
+ Unless required by applicable law or agreed to in writing, software
+ distributed under the License is distributed on an "AS IS" BASIS,
+ WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ See the License for the specific language governing permissions and
+ limitations under the License.
+ */
+
+import UIKit
+import PrebidMobile
+
+class CustomRendererBannerController: NSObject, AdaptedController, PrebidConfigurableBannerController, BannerViewDelegate {
+    
+    var refreshInterval: TimeInterval = 0
+    
+    var prebidConfigId = ""
+
+    var adSizes = [CGSize]()
+    var adFormat: AdFormat?
+    
+    var adBannerView: BannerView?
+    
+    var testPBSSDKConfig = false
+    
+    weak var rootController: AdapterViewController?
+    
+    private let adViewDidReceiveAdButton = EventReportContainer()
+    private let adViewDidFailToLoadAdButton = EventReportContainer()
+    private let adViewWillPresentScreenButton = EventReportContainer()
+    private let adViewDidDismissScreenButton = EventReportContainer()
+    private let adViewWillLeaveApplicationButton = EventReportContainer()
+    
+    private let reloadButton = ThreadCheckingButton()
+    private let stopRefreshButton = ThreadCheckingButton()
+    
+    let lastLoadedAdSizeLabel = UILabel()
+    private let configIdLabel = UILabel()
+    
+    required init(rootController: AdapterViewController) {
+        super.init()
+        self.rootController = rootController
+        Prebid.registerPluginRenderer(SampleCustomRenderer())
+        
+        reloadButton.addTarget(self, action: #selector(reload), for: .touchUpInside)
+        stopRefreshButton.addTarget(self, action: #selector(stopRefresh), for: .touchUpInside)
+        
+        setupAdapterController()
+    }
+    
+    deinit {
+        Targeting.shared.sourceapp = nil
+        Prebid.unregisterPluginRenderer(SampleCustomRenderer())
```

Comment:
I think it is more consistent if you init SampleCustomRenderer() instance in a const instead of passing always new instances. If helps the implementation to be future proof. 
For instance `unregisterPluginRenderer` might be changed and take into account not only name but the instance register.

le sampleCustomRenderer = SampleCustomRenderer() 
Prebid.registerPluginRenderer(sampleCustomRenderer)
Prebid.unregisterPluginRenderer(sampleCustomRenderer)